ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+                    button = el && el[2],
+                    conector = el && el[3],
+                    prevConector = breadcrumbsList[breadcrumbsList.length - 2] &&
+                        breadcrumbsList[breadcrumbsList.length - 2][3];
+
+                // Remove connector from the previous button.
+                prevConector && prevConector.destroy();
+                breadcrumbsList[breadcrumbsList.length - 2].length = 3;
+
+                // Remove SVG elements fromt the DOM.
+                button && button.destroy();
+                conector && conector.destroy();
+            } else {
+                this.updateSingleButton();
+            }
+            breadcrumbsList.pop();
+        }
+    }
+
+    /**
+    * Render the button.
+    *
+    * @requires  modules/breadcrums
+    *
+    * @function Highcharts.Breadcrumbs#renderButton
+    * @param {Highcharts.Breadcrumbs} this
+    *        Breadcrumbs class.
+    * @param {Highcharts.Breadcrumbs} breadcrumb
+    *        Current breadcrumb
+    * @param {Highcharts.Breadcrumbs} posX
+    *        Initial horizontal position
+    * @param {Highcharts.Breadcrumbs} posY
+    *        Initial vertical position
+    * @return {SVGElement|void}
+    *        Returns the SVG button
+    */
+    public renderButton(this: Breadcrumbs,
+        breadcrumb: Array<number|null|Point|SeriesType|SVGAElement>,
+        posX: number, posY: number): SVGElement|undefined {
+        const breadcrumbs = this,
+            chart = this.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = breadcrumbs.options,
+            lang = chart.options.lang;
+
+        if (breadcrumbsOptions && breadcrumb && breadcrumb[1]) {
+            const button = chart.renderer.button(
+                breadcrumbs.defaultFormatter(breadcrumb),
+                posX,
+                posY,
+                function (e: (Event|any)): void {
+                    // Extract events from button object and call
+                    const buttonEvents = breadcrumbsOptions.events && breadcrumbsOptions.events.click;
+                    let callDefaultEvent;
+
+                    if (buttonEvents) {
+                        callDefaultEvent = buttonEvents.call(breadcrumbsOptions, e as any, breadcrumbs);
+                    }
+
+                    // Prevent from click on the current level
+                    if (callDefaultEvent !== false &&
+                        breadcrumbsList[breadcrumbsList.length - 1][1].name !== button.textStr &&
+                        breadcrumbsOptions.showFullPath
+                    ) {
+                        breadcrumbs.multipleDrillUp(breadcrumb[0] as number);
+                    }
+
+                    if (callDefaultEvent !== false && !breadcrumbsOptions.showFullPath) {
+                        breadcrumbs.multipleDrillUp(null as any);
+                    }
+                })
+                .attr({
+                    padding: 3
```

Comment:
Avoid hard coding. In accordance with the range selector and other modules, these attributes should be exposed as `options.buttonTheme`.